generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs

```diff
@@ -40,7 +40,7 @@ export class ItemCountComponent {
    */
   @Input() set params(params: { page?: number; totalItems?: number; itemsPerPage?: number }) {
     if (params.page !== undefined && params.totalItems !== undefined && params.itemsPerPage !== undefined) {
-      this.first = (params.page - 1) * params.itemsPerPage === 0 ? 1 : (params.page - 1) * params.itemsPerPage + 1;
+      this.first = (params.page - 1) * params.itemsPerPage + 1;
       this.second = params.page * params.itemsPerPage < params.totalItems ? params.page * params.itemsPerPage : params.totalItems;
     } else {
```

Comment:
Is this correct for `page: 0`, `totalItems: 0`, `itemsPerPage: 0`?
Is it supposed to happen?

Reply:
This component input parameters can have initial values in the parent component. So yes, probably can have also those values.
In the forms generated by generator there are `*ngIf` attributes which are hiding this component if no data is loaded yet.

Reply:
this.first will be negative for `page: 0`.
And having `itemsPerPage: 0` the item number is not calculable.

IMO
The condition should be changed to 
`if (params.page && params.totalItems !== undefined && params.itemsPerPage) {`

Reply:
Done.